### DISCUSSION TITLE
Is it possible to implement Wireless Protocol in AirSim?

### DISCUSSION THREAD

**Discussion by GeneLYJ** [microsoft/AirSim#4449] [Support Q&A]
Hi, I've tried looking for a way to implement wireless communication among multiple drones but have no result. I'm wondering if there's custom API or any ways to implement wireless communication in AirSim? Many thanks.

**Comment by rajat2004** [comment]
What do you exactly mean by Wireless communication?
As you probably have already seen, APIs allow usage for multiple vehicles, and ROS is a very popular way of working with multiple robots

**Comment by GeneLYJ** [comment]
Wireless Communication as in something similar to Bluetooth. Is ROS widely used in AirSim?

**Comment by xxEoD2242** [comment]
@GeneLYJ I have implemented a custom messaging protocol in AirSim that utilizes the standard mspack system that AirSim is built upon.

If you wanted to do something like Bluetooth, you would need to write the messaging protocols yourself. I got around this during academic research by emulating the drone communications utilizing a matrix and some basic distance metrics. If you're interested, I can send you the fork of AirSim where I did some of this work.

**Comment by GeneLYJ** [comment]
You're using Python to write it? Yes that would be great. Thank you!
